cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -286,7 +286,7 @@ public MetadataChange setProperties(MetadataEntity metadataEntity, Map<String, S
     // if there are more key-value properties then process them updating the final metadata state
     while (iterator.hasNext()) {
       Map.Entry<String, String> next = iterator.next();
-      finalMetadata = setMetadata(new MetadataEntry(metadataEntity, next.getKey(), next.getValue())).getExisting();
+      finalMetadata = setMetadata(new MetadataEntry(metadataEntity, next.getKey(), next.getValue())).getLatest();
```

Comment:
this was an existing bug that the unit tests don't cover, with the consequence that audit logs always lack the last one of the properties that were changed. Unit tests currently only validate the entity ids of the audit events, but not the contents. Unit tests for the SPI will validate this.